From 6088128c5a6aa3c194c1c100f19679894d78f6ab Mon Sep 17 00:00:00 2001
From: Wade Cline <wadecline@hotmail.com>
Date: Fri, 1 Dec 2017 00:15:38 -0800
Subject: [PATCH 3/6] Add OpenSSL rehash messages.

---
 src/modules/extra/m_ssl_openssl.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/modules/extra/m_ssl_openssl.cpp b/src/modules/extra/m_ssl_openssl.cpp
index a6009b1d4..be528d03d 100644
--- a/src/modules/extra/m_ssl_openssl.cpp
+++ b/src/modules/extra/m_ssl_openssl.cpp
@@ -1147,12 +1147,21 @@ class ModuleSSLOpenSSL : public Module
 		if (param != "ssl")
 			return;
 
+		ServerInstance->SNO->WriteGlobalSno('a', (user ? (user->nick + " is r") : "R") + "ehashing OpenSSL module on " + ServerInstance->Config->ServerName);
+		if (user)
+			user->WriteRemoteNotice("*** Rehashing OpenSSL module...");
+
 		try
 		{
 			ReadProfiles();
+			ServerInstance->SNO->WriteGlobalSno('a', "*** Successfully rehashed OpenSSL module on " + ServerInstance->Config->ServerName);
+			if (user)
+				user->WriteRemoteNotice("*** Successfully rehashed OpenSSL module.");
 		}
 		catch (ModuleException& ex)
 		{
+			if (user)
+				user->WriteRemoteNotice("*** Error rehashing OpenSSL module: " + ex.GetReason());
 			ServerInstance->Logs->Log(MODNAME, LOG_DEFAULT, ex.GetReason() + " Not applying settings.");
 		}
 	}
-- 
2.24.1

