From 2c65ab366d9620937ad0e660e4d4bcc8a15bf08c Mon Sep 17 00:00:00 2001
From: Wade Cline <wadecline@hotmail.com>
Date: Fri, 1 Dec 2017 00:15:38 -0800
Subject: [PATCH 3/7] Add OpenSSL rehash messages.

---
 src/modules/extra/m_ssl_openssl.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/modules/extra/m_ssl_openssl.cpp b/src/modules/extra/m_ssl_openssl.cpp
index 7ab7ce258..9e0607922 100644
--- a/src/modules/extra/m_ssl_openssl.cpp
+++ b/src/modules/extra/m_ssl_openssl.cpp
@@ -1168,6 +1168,10 @@ class ModuleSSLOpenSSL : public Module
 		if (!irc::equals(param, "tls") && !irc::equals(param, "ssl"))
 			return;
 
+		ServerInstance->SNO->WriteGlobalSno('a', (user ? (user->nick + " is r") : "R") + "ehashing OpenSSL module on " + ServerInstance->Config->ServerName);
+		if (user)
+			user->WriteRemoteNotice("*** Rehashing OpenSSL module...");
+
 		try
 		{
 			ReadProfiles();
@@ -1175,6 +1179,8 @@ class ModuleSSLOpenSSL : public Module
 		}
 		catch (ModuleException& ex)
 		{
+			if (user)
+				user->WriteRemoteNotice("*** Error rehashing OpenSSL module: " + ex.GetReason());
 			ServerInstance->Logs->Log(MODNAME, LOG_DEFAULT, ex.GetReason() + " Not applying settings.");
 		}
 	}
-- 
2.31.1

