From d920b719562614d9a6b3900aafb12e98fe5001b3 Mon Sep 17 00:00:00 2001
From: Wade Cline <wadecline@hotmail.com>
Date: Fri, 1 Dec 2017 00:15:38 -0800
Subject: [PATCH 4/7] Add OpenSSL rehash messages.

---
 src/modules/extra/m_ssl_openssl.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/modules/extra/m_ssl_openssl.cpp b/src/modules/extra/m_ssl_openssl.cpp
index 43be564c7..5eb6720dc 100644
--- a/src/modules/extra/m_ssl_openssl.cpp
+++ b/src/modules/extra/m_ssl_openssl.cpp
@@ -1249,6 +1249,10 @@ class ModuleSSLOpenSSL : public Module
 		if (!irc::equals(param, "tls") && !irc::equals(param, "ssl"))
 			return;
 
+		ServerInstance->SNO->WriteGlobalSno('a', (user ? (user->nick + " is r") : "R") + "ehashing OpenSSL module on " + ServerInstance->Config->ServerName);
+		if (user)
+			user->WriteRemoteNotice("*** Rehashing OpenSSL module...");
+
 		try
 		{
 			ReadProfiles();
-- 
2.39.3

