From 225f11978d75ca87295e227a56ac5788daa1873c Mon Sep 17 00:00:00 2001
From: Wade Cline <wadecline@hotmail.com>
Date: Sun, 14 Jun 2020 22:28:23 -0700
Subject: [PATCH 5/7] Warn on old client certificate

---
 src/modules/m_sslinfo.cpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/modules/m_sslinfo.cpp b/src/modules/m_sslinfo.cpp
index 70e065257..8d8d5d1f3 100644
--- a/src/modules/m_sslinfo.cpp
+++ b/src/modules/m_sslinfo.cpp
@@ -302,6 +302,16 @@ class ModuleSSLInfo
 			text.append(" and your TLS (SSL) client certificate fingerprint is ").append(cert->GetFingerprint());
 		user->WriteNotice(text);
 
+		// CotSS hack: Warn user if they are using an old client
+		// certificate.
+		if (cert && cert->GetIssuer() == "/CN=frostsnow.net") {
+			user->WriteNotice("WARNING: You are using an old "
+				"client certificate; please update your "
+				"client certificate to the new PKI using the "
+				"instructions located at: "
+				"https://www.frostsnow.net/contact/irc.pdf");
+		}
+
 		if (!cert)
 			return;
 
-- 
2.26.2

