From 7a41d073aae17c1a5d3e5e99f5c3376b9ea83b6f Mon Sep 17 00:00:00 2001
From: Wade Cline <wadecline@hotmail.com>
Date: Fri, 1 Dec 2017 00:15:38 -0800
Subject: [PATCH 3/6] Add OpenSSL rehash messages.

---
 src/modules/extra/m_ssl_openssl.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/modules/extra/m_ssl_openssl.cpp b/src/modules/extra/m_ssl_openssl.cpp
index ec12e3814..d13e49a9d 100644
--- a/src/modules/extra/m_ssl_openssl.cpp
+++ b/src/modules/extra/m_ssl_openssl.cpp
@@ -1154,6 +1154,10 @@ class ModuleSSLOpenSSL : public Module
 		if (!irc::equals(param, "ssl"))
 			return;
 
+		ServerInstance->SNO->WriteGlobalSno('a', (user ? (user->nick + " is r") : "R") + "ehashing OpenSSL module on " + ServerInstance->Config->ServerName);
+		if (user)
+			user->WriteRemoteNotice("*** Rehashing OpenSSL module...");
+
 		try
 		{
 			ReadProfiles();
@@ -1161,6 +1165,8 @@ class ModuleSSLOpenSSL : public Module
 		}
 		catch (ModuleException& ex)
 		{
+			if (user)
+				user->WriteRemoteNotice("*** Error rehashing OpenSSL module: " + ex.GetReason());
 			ServerInstance->Logs->Log(MODNAME, LOG_DEFAULT, ex.GetReason() + " Not applying settings.");
 		}
 	}
-- 
2.24.1

